### PR TITLE
Improve errors in widgets

### PR DESCRIPTION
**Pull request by rubenmoya**
Related to: https://github.com/CartoDB/deep-insights.js/issues/574

screen shot 2017-09-25 at 12 08 19
screen shot 2017-09-25 at 12 09 42
screen shot 2017-09-25 at 12 17 36

## Acceptance steps
- [ ] Should show the timeout error for dataviewmodel and datamodel requests with status `429`
- [ ] With the error view active, should show the widget normal view if the next request is success
- [ ] The refresh button should work for dataviewmodel and datamodel requests
- [ ] Should show the _no data available_ error if there are no points in the bbox
- [ ] WIth the _no data available_ should show the widget normal view if there are points in the bbox
- [ ] Should refresh the data when moving around the map before and after an error

### CODE REVIEW

**Comment by xavijam** [comment]
Tiene pintote.

File: spec/widgets/widget-view-factory.spec.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ describe('widgets/widget-view-factory', function () {
       this.widget = new cdb.core.Model({}, {
         layer: this.layer
       });
+      this.widget.dataviewModel = {};
```
Comment (xavijam) [inline spec/widgets/widget-view-factory.spec.js]: Not need anything else?

File: src/widgets/formula/placeholder.tpl [hunk 2]
```diff
@@ -0,0 +1 @@
+<div class="CDB-Widget-listItem--fake"></div>
```
Comment (xavijam) [inline src/widgets/formula/placeholder.tpl]: Eing? Didn't we have one of these?

File: src/widgets/widget-error-view.js [hunk 3]
```diff
@@ -1,49 +1,56 @@
...
-
-  show: function () {
-    this.$el.removeClass('is-hidden');
+    this._model.refresh();
   },
 
-  hide: function () {
-    this.$el.addClass('is-hidden');
+  _reset: function () {
+    this.$el.removeClass('CDB-Widget--error');
```
Comment (xavijam) [inline src/widgets/widget-error-view.js]: Do you know you can remove several classes in the same instance?

File: src/widgets/widget-view.js [hunk 4]
```diff
@@ -15,26 +24,71 @@ module.exports = cdb.core.View.extend({
 
   initialize: function () {
     this.listenTo(this.model, 'destroy', this.clean);
```
Comment (xavijam) [inline src/widgets/widget-view.js]: I'd create:
```js
var dataviewModel = this.model.dataviewModel;
this.listenTo(dataviewModel, 'sync change:data', this._onDataChanged);
if (dataviewModel && dataviewModel._totals) {
  …
}
```

**Comment by xavijam** [comment]
Ok, tested in staging and I've found a problem trying to refresh a histogram widget (refresh is undefined). Back to development!

File: spec/widgets/widget-view-factory.spec.js (hunk 1 above)
Comment (rubenmoya) [reply spec/widgets/widget-view-factory.spec.js]: No, just needed because of `dataviewModel._totals` would be `undefined.totals` otherwise

**Comment by rubenmoya** [comment]
retest this please

**Pull request merged**